### ISSUE TITLE
Microsoft.CodeAnalysis.CSharp.Emit.UnitTests failing frequently on CI with StackOverflowException

### ISSUE THREAD

**Issue by mavasani** [dotnet/roslyn#45558]
```
Errors Microsoft.CodeAnalysis.CSharp.Emit.UnitTests.dll

Process is terminated due to StackOverflowException.
Command: C:\Users\vsagent\.nuget\packages\xunit.runner.console\2.4.1-pre.build.4059\tools\net472\xunit.console.x86.exe "F:\workspace\_work\1\s\artifacts\bin\Microsoft.CodeAnalysis.CSharp.Emit.UnitTests\Release\net472\Microsoft.CodeAnalysis.CSharp.Emit.UnitTests.dll" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.CodeGenRefLocalTests" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.CodeGenRefReadOnlyReturnTests" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.CodeGenStackAllocInitializerTests" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.DesktopStrongNameProviderTests" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.Emit.CodeGenScriptTests" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.Emit.CompilationEmitTests" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.Emit.CompilationEmitTests+Match" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.Emit.CompilationEmitTests+TestPEBuilder" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.Emit.CompilationEmitTests+TestResourceSectionBuilder" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.Emit.DeterministicTests" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.Emit.DeterministicTests+WriteOnlyStream" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.Emit.EmitCustomModifiers" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.Emit.EndToEndTests" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.Emit.EntryPointTests" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.Emit.InAttributeModifierTests" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.Emit.NoPiaEmbedTypes" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.Emit.OptionalArgumentsTests" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.Emit.Perf" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.Emit.ResourceTests" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.Emit.UnmanagedTypeModifierTests" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.EmitErrorTests" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.EmitMetadata" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.InternalsVisibleToAndStrongNameTests" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.PDB.CheckSumTest" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.PDB.CSharpDeterministicBuildCompilationTests" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.PDB.CSharpPDBTestBase" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.PDB.PDBAsyncTests" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.PDB.PDBConstantTests" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.PDB.PDBDynamicLocalsTests" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.PDB.PDBEmbeddedSourceTests" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.PDB.PDBIteratorTests" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.PDB.PDBLambdaTests" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.PDB.PDBLocalFunctionTests" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.PDB.PDBSourceLinkTests" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.PDB.PDBTests" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.PDB.PDBTupleTests" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.PDB.PDBUsingTests" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.PDB.PDBWinMdExpTests" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.PDB.PortablePdbTests" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.PrivateProtected" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.ReadOnlyStruct" -class "Microsoft.CodeAnalysis.CSharp.UnitTests.RefReturnTests"  -xml "F:\workspace\_work\1\s\artifacts\TestResults\Release\Microsoft.CodeAnalysis.CSharp.Emit.UnitTests.dll.4.xml" -noshadow -verbose
xUnit output log: F:\workspace\_work\1\s\artifacts\log\Release\xUnitFailure-Microsoft.CodeAnalysis.CSharp.Emit.UnitTests.dll.4.log

Process is terminated due to StackOverflowException.
================
```

https://dev.azure.com/dnceng/public/_build/results?buildId=710863&view=logs&j=ccc4f4cd-3db2-535a-b197-b0dc972cf7f6&t=b5751933-4a09-5014-1b76-81e74adfd51c

https://dev.azure.com/dnceng/public/_build/results?buildId=711157&view=logs&j=ccc4f4cd-3db2-535a-b197-b0dc972cf7f6&t=b5751933-4a09-5014-1b76-81e74adfd51c

https://dev.azure.com/dnceng/public/_build/results?buildId=710863&view=logs&j=ccc4f4cd-3db2-535a-b197-b0dc972cf7f6&t=b5751933-4a09-5014-1b76-81e74adfd51c

**Comment by mavasani** [comment]
@dotnet/roslyn-compiler @dotnet/roslyn-infrastructure Is this a known issue?

**Comment by jaredpar** [comment]
Yes @RikkiGibson just submitted a fix last night for it: #45554

**Comment by mavasani** [comment]
Thanks!
